QA_PROTOCOL.rst

```diff
@@ -0,0 +1,54 @@
+..
+    Copyright 2017-2019 - Swiss Data Science Center (SDSC)
+    A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+    Eidgenössische Technische Hochschule Zürich (ETHZ).
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+==============================
+ Renku Python QA Protocol
+==============================
+
+This document describes smoke tests for checking that a new ``renku-python``
+release works with basic functionality.
+
+Setup & general
+---------------
+
+- [ ] Check installation via pip.
+- [ ] Check output of ``renku --version``.
+- [ ] Check that integration tests run successfully.
+- [ ] Look at the docs and make sure they're up to date.
+- [ ] Check that ``--help`` is up to date for changed commands.
+
+Test new repository
+-------------------
+- [ ] Init renku in a bare repo
+- [ ] Create a dataset and add a file
+- [ ] List datasets and files
+- [ ] Import a dataset from Zenodo and/or Dataverse
+- [ ] List datasets and files
+- [ ] Remove a dataset
+- [ ] ``renku run`` a simple command, e.g. ``cp``
```

Comment:
maybe add that this should have more than one component? I always do something like this

```
renku run echo "bla" > bla
renku run wc bla > bla.wc
renku run echo "bla2" > bla
renku status # should show that bla.wc is outdated
renku update # should update bla
```

Reply:
however, isn't this what the tests are doing already anyway? Or in which cases would this manual check be needed even if the tests are passing?

Reply:
Instead of a telling to QA explicit commands to execute (and like you pointed out, which tests are doing it anyway), tell them a use case they should replicate/achieve and let them figure out how to do it. That way you always have a human component in breaking things more efficiently. :smile: 

So, an example of a defined case would be: Using renku create a 3 step workflow process which counts the words in a file and writes them into another file. Create a new file and rerun the pipeline to see the new results.

Reply:
I remember renku failing less than a week ago despite all the tests passing :wink: 

The point is that yes, this is what the tests are doing anyways, but in isolation. Doing everything in a single renku repo with different steps etc. might show issues that single tests don't catch but a human catches immediately.